plumbing/transport/receive_pack.go

```diff
@@ -160,10 +175,6 @@ func sendReportStatus(w io.WriteCloser, unpackErr error, cmdStatus map[plumbing.
 		return err
 	}
 
-	if err := w.Close(); err != nil {
```

Comment:
It looks like this change stops the `writer` being closed when `unpackErr != nil` on line `130`.